### PR TITLE
Improving Log UI output text colors

### LINKED ISSUE #186: Improve Log UI Component
add support of :
- log text colouration depending on message type (info, error, etc..)
- clear text feature
- copy to clipboard feature

### PR DESCRIPTION
**Pull request by JFancis**
I created DrawColoredTextLine to format the text color and add conditions for different cases (error, warning...).

### CODE REVIEW

File: Tetragrama/src/Components/LogUIComponent.cpp [hunk 1]
```diff
@@ -26,11 +28,47 @@ namespace Tetragrama::Components {
...
 
+
+            ImGuiListClipper clipper;
+            clipper.Begin(m_content_line_offsets.Size);
+            while (clipper.Step()) {
+                for (int line_no = clipper.DisplayStart; line_no < clipper.DisplayEnd; line_no++) {
+                    const char* line_start = buf + m_content_line_offsets[line_no];
+                    const char* line_end   = (line_no + 1 < m_content_line_offsets.Size) ? (buf + m_content_line_offsets[line_no + 1] - 1) : buf_end;
+                   
+                    ImVec4 color;
```
Comment (JeanPhilippeKernel) [inline Tetragrama/src/Components/LogUIComponent.cpp]: We don't need those `color `and `has_color `variables

File: Tetragrama/src/Components/LogUIComponent.cpp [hunk 2]
```diff
@@ -26,11 +28,47 @@ namespace Tetragrama::Components {
...
+                    if (strstr(line_start, "[warning]")) {
+                        Helpers::DrawColoredTextLine(line_start, line_end, ImVec4(1.0f, 0.87f, 0.37f, 1.f));                       
+                    }
+                    if (strstr(line_start, "[debug]")) {
+                        Helpers::DrawColoredTextLine(line_start, line_end, ImVec4(0.94f, 0.39f, 0.13f, 1.0f));
+                    }                
+                    if (strstr(line_start, "[info]")) {
+                        Helpers::DrawColoredTextLine(line_start, line_end, ImVec4(0.46f, 0.96f, 0.46f, 1.f));
+                    }
+                    if (strstr(line_start, "[trace]")) {
```
Comment (JeanPhilippeKernel) [inline Tetragrama/src/Components/LogUIComponent.cpp:52]: why 2 definitions of [trace]?

File: Tetragrama/src/Components/LogUIComponent.cpp [hunk 3]
```diff
@@ -26,11 +28,47 @@ namespace Tetragrama::Components {
         ImGui::PushStyleVar(ImGuiStyleVar_ItemSpacing, ImVec2(0, 0));
 
         if (m_content.size() > 0) {
+
             const char* buf     = m_content.begin();
             const char* buf_end = m_content.end();
-            ImGui::TextUnformatted(buf, buf_end);
-        }
 
+
```
Comment (JeanPhilippeKernel) [inline Tetragrama/src/Components/LogUIComponent.cpp]: remove this line

File: Tetragrama/src/Helpers/UIComponentDrawerHelper.cpp [hunk 4]
```diff
@@ -448,4 +448,11 @@ namespace Tetragrama::Helpers {
         ImGui::Columns(1);
         ImGui::PopID();
     }
+
+    void DrawColoredTextLine(const char* start, const char* end, const ImVec4& color) {
+
```
Comment (JeanPhilippeKernel) [inline Tetragrama/src/Helpers/UIComponentDrawerHelper.cpp]: remove this line

File: Tetragrama/src/Components/LogUIComponent.cpp [hunk 5]
```diff
@@ -26,11 +28,47 @@ namespace Tetragrama::Components {
         ImGui::PushStyleVar(ImGuiStyleVar_ItemSpacing, ImVec2(0, 0));
 
         if (m_content.size() > 0) {
+
```
Comment (JeanPhilippeKernel) [inline Tetragrama/src/Components/LogUIComponent.cpp]: remove this empty line

File: Tetragrama/src/Components/LogUIComponent.cpp [hunk 6]
```diff
@@ -12,6 +13,7 @@ namespace Tetragrama::Components {
         return false;
     }
 
+
```
Comment (JeanPhilippeKernel) [inline Tetragrama/src/Components/LogUIComponent.cpp]: remove this empty line

File: Tetragrama/src/Components/LogUIComponent.cpp (hunk 1 above)
Comment (JFancis) [reply Tetragrama/src/Components/LogUIComponent.cpp]: okey

File: Tetragrama/src/Components/LogUIComponent.cpp (hunk 2 above)
Comment (JFancis) [reply Tetragrama/src/Components/LogUIComponent.cpp:52]: it is an omission

File: Tetragrama/src/Components/LogUIComponent.cpp (hunk 3 above)
Comment (JFancis) [reply Tetragrama/src/Components/LogUIComponent.cpp]: well noted

File: Tetragrama/src/Helpers/UIComponentDrawerHelper.cpp (hunk 4 above)
Comment (JFancis) [reply Tetragrama/src/Helpers/UIComponentDrawerHelper.cpp]: well noted

File: Tetragrama/src/Components/LogUIComponent.cpp (hunk 5 above)
Comment (JFancis) [reply Tetragrama/src/Components/LogUIComponent.cpp]: well noted

File: Tetragrama/src/Components/LogUIComponent.cpp (hunk 6 above)
Comment (JFancis) [reply Tetragrama/src/Components/LogUIComponent.cpp]: well noted

File: Tetragrama/src/Components/LogUIComponent.cpp (hunk 1 above)
Comment (JFancis) [reply Tetragrama/src/Components/LogUIComponent.cpp]: done

File: Tetragrama/src/Components/LogUIComponent.cpp (hunk 2 above)
Comment (JFancis) [reply Tetragrama/src/Components/LogUIComponent.cpp:52]: done - it part to critical

File: Tetragrama/src/Components/LogUIComponent.cpp (hunk 3 above)
Comment (JFancis) [reply Tetragrama/src/Components/LogUIComponent.cpp]: done

File: Tetragrama/src/Helpers/UIComponentDrawerHelper.cpp (hunk 4 above)
Comment (JFancis) [reply Tetragrama/src/Helpers/UIComponentDrawerHelper.cpp]: done

File: Tetragrama/src/Components/LogUIComponent.cpp (hunk 6 above)
Comment (JFancis) [reply Tetragrama/src/Components/LogUIComponent.cpp]: done

**Review by JeanPhilippeKernel — APPROVED** [review]

**Comment by JeanPhilippeKernel** [comment]
Thanks for this awesome work 🎉!

**Pull request merged**